File: Interviews/LeetCode/Daily-Problems/2022/October/Day-18/38.cpp
class Solution {
public:
    string spellCount(int n) {
        string ans = "";
        while (n > 0) {
            ans += (n % 10) + '0';
            n /= 10;
        }

        reverse(ans.begin(), ans.end());
        return ans;
    }

    string convertToShit(string s) {
        int cnt = 1;
        string ans = "";
        char cur = s[0];
        for (int i = 1; i < s.size(); i++) {
            if (s[i] == cur) {
                cnt += 1;
            } else {
                ans += spellCount(cnt) + cur;
                cur = s[i];
                cnt = 1;
            }
        }

        return ans;
    }

    string countAndSay(int n) {
        string ans = "1";
        for (int i = 2; i <= n; i++) {
            ans = convertToShit(ans + 'a');
        }

        return ans;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-18/38.py
class Solution:
    def convertToShit(self, s: string) -> str:
        cnt, ans, cur = 0, "", s[0]
        for i in s:
            if i == cur:
                cnt += 1
            else:
                ans += str(cnt) + cur
                cur, cnt = i, 1
        
        return ans
    
    def countAndSay(self, n: int) -> str:
        ans = "1"
        for i in range(1, n):
            ans = self.convertToShit(ans + 'a')
        return ans